src/webapi/selection.rs

```diff
@@ -0,0 +1,353 @@
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/collapseToStart)
+    pub fn collapse_to_start(&self) -> Result<(), InvalidStateError> {
+        js_try! ( @(no_return)
+            @{self}.collapseToStart();
+        ).unwrap()
+    }
+
+    /// Collapses the [Selection](struct.Selection.html) to the end of the last range in the
+    /// selection. If the content is focused or editable, the caret will blink there.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/collapseToEnd)
+    pub fn collapse_to_end(&self) -> Result<(), InvalidStateError> {
+        js_try! ( @(no_return)
+            @{self}.collapseToEnd();
+        ).unwrap()
+    }
+
+    /// Moves the focus of the selection to a specified point. The anchor of the selection does not
+    /// move. The selection will be from the anchor node to the new focus regardless of direction.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/extend)
+    pub fn extend<N: INode>(&self, node: &N, offset: Option<u32>) -> Result<(), InvalidStateError> {
+        js_try! ( @(no_return)
+            @{self}.extend(@{node.as_ref()}, @{offset});
+        ).unwrap()
+    }
+
+    /// Sets the selection to be a range including all or parts of the two specified
+    /// [Node](struct.Node.html)s, and any content located between them.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/setBaseAndExtent)
+    pub fn set_base_and_extent<N: INode>(&self, anchor_node: &N, anchor_offset: u32, focus_node: &N, focus_offset: u32) -> Result<(), InvalidStateError> {
+        js_try! ( @(no_return)
+            @{self}.setBaseAndExtent(@{anchor_node.as_ref()}, @{anchor_offset}, @{focus_node.as_ref()}, @{focus_offset});
+        ).unwrap()
+    }
+
+    /// Adds all the children of the specified [Node](struct.Node.html) to the selection. Previous
+    /// selection is lost.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/selectAllChildren)
+    pub fn select_all_children<N: INode>(&self, node: &N) {
+        js! { @(no_return)
+            @{self}.selectAllChildren(@{node.as_ref()});
+        };
+    }
+
+    /// Deletes the actual text being represented by the [Selection](struct.Selection.html) from
+    /// the document's DOM.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/deleteFromDocument)
+    pub fn delete_from_document(&self) {
+        js! { @(no_return)
+            @{self}.deleteFromDocument();
+        };
+    }
+
+    /// Indicates if the node is part of the selection.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/containsNode)
+    pub fn contains_node<N: INode>(&self, node: &N, allow_partial_containment: bool) -> bool {
```

Comment:
As much as I like to usually leave APIs as they're defined originally this boolean is quite nasty; perhaps we should make two methods out of this one? `contains_whole` and `contains_part_of` perhaps?